src/c/c_client.cpp

```diff
@@ -30,517 +30,654 @@
 
 using namespace SmartRedis;
 
+
+// Return a pointer to a new Client.
+// The caller is responsible for deleting the client via DeleteClient().
 extern "C"
-void* SmartRedisCClient(bool cluster)
+void *SmartRedisCClient(bool cluster)
 {
-  /* Return a pointer to a new Client.
-  The user is responsible for deleting the client.
-  */
-  Client* s = new Client(cluster);
-  return (void*)s;
+  Client *s = NULL;
+
+  try {
+    s = new Client(cluster);
+  } catch (...) {
+    s = NULL;
+  }
+
+  return reinterpret_cast<void *>(s);
 }
 
+// Free the memory associated with the c client.
 extern "C"
-void DeleteCClient(void* c_client)
+void DeleteCClient(void *c_client)
 {
-  /* This function frees the memory associated
-  with the c client.
-  */
-  Client* s = (Client*)c_client;
+  // Sanity check params
+  if (NULL == c_client)
```

Comment:
Probably would get rid of unnecessary mid-function return with a ``if (NULL != c_client)``

Reply:
See previous comment -- I like to keep the sanity checks separate